modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/date/facet/portlet/DateFacetPortlet.java

```diff
@@ -0,0 +1,242 @@
+import com.liferay.portal.search.web.internal.display.context.PortletRequestThemeDisplaySupplier;
+import com.liferay.portal.search.web.internal.display.context.ThemeDisplaySupplier;
+import com.liferay.portal.search.web.portlet.shared.search.PortletSharedSearchRequest;
+import com.liferay.portal.search.web.portlet.shared.search.PortletSharedSearchResponse;
+
+import java.io.IOException;
+
+import java.util.Hashtable;
+import java.util.Map;
+
+import javax.portlet.Portlet;
+import javax.portlet.PortletException;
+import javax.portlet.RenderRequest;
+import javax.portlet.RenderResponse;
+
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Petteri Karttunen
+ */
+@Component(
+	property = {
+		"com.liferay.portlet.add-default-resource=true",
+		"com.liferay.portlet.css-class-wrapper=portlet-date-facet",
+		"com.liferay.portlet.display-category=category.hidden",
+		"com.liferay.portlet.header-portlet-css=/css/main.css",
+		"com.liferay.portlet.icon=/icons/search.png",
+		"com.liferay.portlet.instanceable=true",
+		"com.liferay.portlet.layout-cacheable=true",
+		"com.liferay.portlet.preferences-owned-by-group=true",
+		"com.liferay.portlet.private-request-attributes=false",
+		"com.liferay.portlet.private-session-attributes=false",
+		"com.liferay.portlet.restore-current-view=false",
+		"com.liferay.portlet.use-default-template=true",
+		"javax.portlet.display-name=Date Facet",
+		"javax.portlet.expiration-cache=0",
+		"javax.portlet.init-param.template-path=/META-INF/resources/",
+		"javax.portlet.init-param.view-template=/date/facet/view.jsp",
+		"javax.portlet.name=" + DateFacetPortletKeys.DATE_FACET,
+		"javax.portlet.resource-bundle=content.Language",
+		"javax.portlet.security-role-ref=guest,power-user,user",
+		"javax.portlet.version=3.0"
+	},
+	service = Portlet.class
+)
+public class DateFacetPortlet extends MVCPortlet {
+
+	@Override
+	public void render(
+			RenderRequest renderRequest, RenderResponse renderResponse)
+		throws IOException, PortletException {
+
+		PortletSharedSearchResponse portletSharedSearchResponse =
+			_portletSharedSearchRequest.search(renderRequest);
+
+		DateFacetDisplayContext dateFacetDisplayContext = _buildDisplayContext(
+			portletSharedSearchResponse, renderRequest);
+
+		if (dateFacetDisplayContext.isRenderNothing()) {
+			renderRequest.setAttribute(
+				WebKeys.PORTLET_CONFIGURATOR_VISIBILITY, Boolean.TRUE);
+		}
+
+		renderRequest.setAttribute(
+			WebKeys.PORTLET_DISPLAY_CONTEXT, dateFacetDisplayContext);
+
+		super.render(renderRequest, renderResponse);
+	}
+
+	@Activate
+	protected void activate(Map<String, Object> properties) {
+		if (FeatureFlagManagerUtil.isEnabled("LPS-153839")) {
```

Comment:
when i set the FF to true im getting some kind of infinite loop here... the activate method is continually called, maybe triggered by configuration.update()? and the widget doesnt appear in the "Add" panel, so i cant test it on a page. did you see anything like this with your testing?

Reply:
worked around this setting FF to false and adding directly to category.search, but when saving the widget configuration i get `java.lang.IllegalArgumentException: Invalid range syntax [past-hour TO *]`

Reply:
Interesting, didn't have issues with the activate(). @joshchong ?

The validation issue was a copy&paste issue. Fixed.